tests/unit_tests/test_sky_import.py

```diff
@@ -0,0 +1,106 @@
+"""Ensure the lazy import of modules is not reverted.
+
+This test ensures that the lazy import of modules is not reverted.
+"""
+import sys
+
+import pytest
+
+
+def test_sky_import():
+    # Clean modules that are lazy imported by sky
+    modules_to_clean = [
+        module for module in list(sys.modules.keys()) if any(
+            module.startswith(prefix) for prefix in [
+                'numpy',
+                'pendulum',
+                'cryptography',
+                'requests',
+                'jsonschema',
+                'psutil',
+                'jinja2',
+                'yaml',
+                'rich.console',
+                'rich.progress',
+                'networkx',
+                'httpx',
+                'pydantic',
+            ])
+    ]
+
+    for module in modules_to_clean:
+        del sys.modules[module]
+
+    # Import sky
+    try:
+        import sky
+    except Exception as e:
+        print(f"Failed to import sky: {e}")
+        raise
+
+    # Check that the lazy imported modules are not imported
+    assert 'requests.packages.urllib3.util.retry' not in sys.modules
+    assert 'requests.adapters' not in sys.modules
+    assert 'requests' not in sys.modules
+    assert 'yaml' not in sys.modules
+    assert 'cryptography.hazmat.backends' not in sys.modules
+    assert 'cryptography.hazmat.primitives' not in sys.modules
+    assert 'cryptography.hazmat.primitives.asymmetric' not in sys.modules
+    assert 'rich.progress' not in sys.modules
+    assert 'rich.console' not in sys.modules
+    assert 'httpx' not in sys.modules
+    assert 'psutil' not in sys.modules
+    assert 'networkx' not in sys.modules
+    assert 'numpy' not in sys.modules
+    assert 'jinja2' not in sys.modules
+    assert 'pydantic' not in sys.modules
+    assert 'jsonschema' not in sys.modules
+    assert 'pendulum' not in sys.modules
+
+    # Check that the lazy imported modules are imported after used
```

Comment:
Not sure whether the following assertions are behaviors we want to keep. To me, what modules are used by the function seems like an easy to change implementation detail. Maybe existing unit tests of these functions themselves would be sufficient to ensure they still work after this PR.

Reply:
Make sense, I will remove the check below.